src/stream/src/executor/mview/materialize.rs

```diff
@@ -223,10 +222,9 @@ impl<S: StateStore, SD: ValueRowSerde> MaterializeExecutor<S, SD> {
                     }
                 }
                 Message::Barrier(b) => {
-                    let mutation = b.mutation.clone();
                     // If a downstream mv depends on the current table, we need to do conflict check again.
                     if !self.state_table.is_consistent_op()
-                        && Self::new_downstream_created(mutation, self.actor_context.id)
+                        && b.has_new_downstream(self.actor_context.id)
```

Comment:
This might have some conflicts with this PR `refactor(state-table): make consistent op an enum in state table (#16471)`. Make sure to resolve it later.